### ISSUE TITLE
gitlab_monitor got renamed to gitlab_exporter in 12.3

### ISSUE THREAD

**Issue by baurmatt** [voxpupuli/puppet-gitlab#323]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: All
- Ruby: All
- Distribution: All
- Module version: Current

## How to reproduce (e.g Puppet code you use)

Read changelog ;) --> https://docs.gitlab.com/omnibus/update/gitlab_12_changes.html#123

Deprecated. Will be removed in 13.

## What are you seeing

gitlab_exporter got renamed to gitlab_monitor.

## What behaviour did you expect instead

sys11gitlab::gitlab_exporter should be available

**Comment by baurmatt** [comment]
I'm not sure how we handle deprecated/renamed parameters. Just add an additional parameter `gitlab_exporter` and throw a `warning()` for `sys11gitlab::gitlab_monitor`?

Happy to implement this after we figured out how to handle the deprecation! :)

**Comment by LongLiveCHIEF** [comment]
We've had similar issues come up a few times before. (Hence the section in the README about what major/minor versions this puppet module targets for support).

The suggestion you had is one approach we've used, and that works well. (In the past I think we used a `notify` resource, but the last time this happened we were still only supporting puppet 3/4). 
For an example, check out how we handled the change of the `skip_auto_migrations` config key to `skip_auto_reconfigure` (see: #234 ) 

The other approach we've used is to coerce one setting into another, like we did with the `edition` parameter this module used to support. (see [manifests/install.pp](manifests/install.pp))

The first case is used when _gitlab_ changes or removes a parameter, and the latter is used when we deprecate or remove a parameter that was exclusive to this module.
